internal/serialization/serialization_test.go

```diff
@@ -365,3 +365,95 @@ func mustData(value interface{}, err error) iserialization.Data {
 	}
 	return value.(iserialization.Data)
 }
+
+type student struct {
+	Name *string
+	Age  int32
+}
+
+type studentCompactSerializer struct{}
+
+func (studentCompactSerializer) Type() reflect.Type {
+	return reflect.TypeOf(student{})
+}
+
+func (s studentCompactSerializer) TypeName() string {
+	return s.Type().Name()
```

Comment:
This is probably not the place to comment on this, but I am afraid that a user can have serializer conflicts following this kind of tests/examples. `reflect.Type::Name()` does not return a unique string. I think should try to document that anywhere we use this.

Reply:
What about always writing the typename explicitly like "student"

Reply:
IMO that's better!

Reply:
Done